systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java

```diff
@@ -0,0 +1,163 @@
+
+        assertThatRollingUpdatedFinished(KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME), KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        TimeMeasuringSystem.stopOperation(rollingUpdateOperation);
+        TimeMeasuringSystem.stopOperation(operationID);
+    }
+
+    @Test
+    void testRecoveryDuringKafkaRollingUpdate() {
+        // @TODO add send-recv messages during this test
+        operationID = startTimeMeasuring(Operation.CLUSTER_RECOVERY);
+
+        String firstKafkaPodName = KafkaResources.kafkaPodName(CLUSTER_NAME, 0);
+        String logKafkaPattern = "'Exceeded timeout of .* while waiting for Pods resource " + firstKafkaPodName + "'";
+
+        resources().kafkaEphemeral(CLUSTER_NAME, 3).done();
+
+        LOGGER.info("Update resources for pods");
+
+        resources().kafkaEphemeral(CLUSTER_NAME, 3)
+                .editSpec()
+                .editKafka()
+                .withNewResources()
+                .withNewRequests()
+                .withMilliCpu("100000m")
+                .endRequests()
+                .endResources()
+                .endKafka()
+                .endSpec()
+                .done();
+
+        TestUtils.waitFor("Wait till rolling update of pods start", 2000, CO_OPERATION_TIMEOUT,
+            () -> !CLIENT.pods().withName(firstKafkaPodName).isReady());
+
+        TestUtils.waitFor("Wait till rolling update timeouted", 2000, CO_OPERATION_TIMEOUT + 20000,
+            () -> !KUBE_CLIENT.searchInLog("deploy", "strimzi-cluster-operator", TimeMeasuringSystem.getCurrentDuration(testClass, testName, operationID), logKafkaPattern).isEmpty());
+
+        assertThatRollingUpdatedFinished(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME));
+
+        LOGGER.info("Wait till another rolling update starts");
+        // Second part
+        String rollingUpdateOperation = TimeMeasuringSystem.startOperation(Operation.ROLLING_UPDATE);
+
+        TestUtils.waitFor("Wait till rolling update timeouted", 2000, 180000,
+                () -> !KUBE_CLIENT.searchInLog("deploy", "strimzi-cluster-operator", TimeMeasuringSystem.getCurrentDuration(testClass, testName, rollingUpdateOperation), logKafkaPattern).isEmpty());
+
+        assertThatRollingUpdatedFinished(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), KafkaResources.zookeeperStatefulSetName(CLUSTER_NAME));
+
+        TimeMeasuringSystem.stopOperation(rollingUpdateOperation);
+        TimeMeasuringSystem.stopOperation(operationID);
+    }
+
+    void assertThatRollingUpdatedFinished(String rolledComponent, String stableComponent) {
+        assertThat(rolledComponent + "-1 is not in ready state", CLIENT.pods().withName(rolledComponent + "-1").isReady());
+        assertThat(rolledComponent + "-2 is not in ready state", CLIENT.pods().withName(rolledComponent + "-2").isReady());
```

Comment:
These aren't completely correct when ZK is the `rolledComponent`, because CO will roll leader last if pod 0 is leader it would roll pod 1, so -0 and -2 would still be ready. Perhaps you want to change the logic so you don't care which is particular pod is ready or not, but are asserting on the number of pods which are ready.